code/game/objects/items/rcd/RHD.dm

```diff
@@ -201,18 +201,18 @@
 	return data
 
 /obj/item/construction/proc/toggle_silo(mob/user)
+	. = TRUE
```

Comment:
Why did you change this? This is worse than the code before it.

Reply:
I made an wrong assumption previously. The return value of this proc should always be true even if it doesn't succeed because ui_act() will check that to determine if the action was handled or not and so it's not important if the action was successful or not

This is the parent proc for all rcd like devices 
https://github.com/tgstation/tgstation/blob/e16588b682a8a1516790d00ff85b305fcc62c11d/code/game/objects/items/rcd/RHD.dm#L217-L223

So since toggle_silo() now always returns true take the rcd for example.  It can now early return here
https://github.com/tgstation/tgstation/blob/e16588b682a8a1516790d00ff85b305fcc62c11d/code/game/objects/items/rcd/RCD.dm#L463-466.

If it did return false instead when the silo link failed then it would waste time going through the remaining if conditions

Reply:
`ui_act`'s return value isn't if it succeeded, it's if the UI should update. Sometimes it's reasonable to return FALSE, preferences does that a lot.